darts/timeseries.py

```diff
@@ -2796,6 +2796,32 @@ def drop_columns(self, col_names: Union[List[str], str]) -> "TimeSeries":
         new_xa = self._xa.drop_sel({"component": col_names})
         return self.__class__(new_xa)
 
+    def keep_columns(self, col_names: Union[List[str], str]) -> "TimeSeries":
+        """
+        Return a new ``TimeSeries`` instance containing only a subset of the columns/components.
+
+        Parameters
+        -------
+        col_names
+            String or list of strings corresponding to the columns to be kept.
+
+        Returns
+        -------
+        TimeSeries
+            A new TimeSeries instance with only the specified columns.
+        """
+        if isinstance(col_names, str):
+            col_names = [col_names]
+
+        raise_if_not(
+            all([(x in self.columns.to_list()) for x in col_names]),
+            "Some column names in col_names don't exist in the time series.",
+            logger,
+        )
+
+        new_xa = self._xa.sel({"component": col_names})
+        return self.__class__(new_xa)
+
```

Comment:
hello, is this function equivalent to selecting the component names from the series as in the following code : 
```
from darts import TimeSeries
from darts.datasets import AirPassengersDataset

univ_series = AirPassengersDataset().load()
multi_series = univ_series.stack(univ_series + 200)
stoch_series = multi_series.concatenate(multi_series, axis = 2) # stochastic, with sample = 2

stoch_series['#Passengers'] # should give a series of 1 component and sample = 2
```

Reply:
Nice trick! I did not know that it was possible to use the slicing operator to extract subset of the columns/components. It certainly makes this function useless, I'll remove it in my next commit.

Reply:
check first before deleting if it makes more sense to do it like you are doing it, your method might be efficient memory wise, as it might not be copying the data. I guess the Xarray.sel() is not copying but then if a new TimeSeries is created I am not sure what would happen. you might want to check if the data is being copied or just a view is provided on the underlying numpy.

Reply:
After looking at the source code of the `__getitem__` operator, it appears that it is also using the `xarray.sel`method to create and return a new `TimeSeries`. The two differences are that the `__getitem__` method makes sure to drop any existing hierarchy whereas my method `keep_columns` verify that the columns asked are indeed in the original `TimeSeries`. Unless we want to explicitly have such a function, it would probably be better to remove my function and use the section I wrote to also showcase the fact that the slicing operator can also be applied to components.

Reply:
+1 for dropping this one, I think `__getitem__` is sufficient. It does return a new TimeSeries containing its own copy of the data, but as should ALL TimeSeries methods :) This is required to keep our TimeSeries immutable and avoid lots and lots of future pain.